x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/hooks/form.tsx

```diff
@@ -360,7 +366,7 @@ export function useOnSubmit({
     setValidationResults,
     hasAgentPolicyError,
     setHasAgentPolicyError,
-    isInitialized: isInitializedRef.current,
+    isInitialized,
```

Comment:
sometimes `validationErrors` and `packagePolicy` goes out of sync: ![Screen Shot 2023-01-30 at 19 46 02](https://user-images.githubusercontent.com/20814186/215554577-c7ede171-34ff-44d1-9dc6-1678d84abb5d.png) 

this happens when our extension component fires an effect on mount to initialize the package policy before fleet's effect (`init()`)

making this property stateful makes the extension component render only after fleet's effect has fired, which sorts out the ordering. 

still kinda odd they are not in sync tho, even if fleet's effect fires last, i'd expect the policy to be changed as well. you can check the logs and reproduce this on [env](https://kibana-pr-149649.kb.us-west2.gcp.elastic-cloud.com:9243/app/fleet/integrations/cloud_security_posture/add-integration/cspm) by going to the install page -> canceling and going in again

@nchaulet